perma_web/tasks/dev.py

```diff
@@ -701,3 +707,496 @@ def populate_folder_cached_path(ctx, batch_size=500):
+    sync registrars/orgs and then delete the other ones.
+
+    If the account with the links is not the one they have logged into most recently,
+    move the links to the most recently logged into account, sync the registrars/orgs,
+    and then delete the others.
+    """
+    [account_with_links] = filter(lambda u: u.link_count, user_list)
+    most_recently_logged_into_account = account_with_links
+    for user in user_list:
+        if user.last_login and user.last_login > most_recently_logged_into_account.last_login:
+            most_recently_logged_into_account = user
+
+    if account_with_links == most_recently_logged_into_account:
+        to_keep = account_with_links
+        to_delete = list(filter(lambda u: u is not to_keep, user_list))
+        merge_accounts(to_keep, to_delete)
+    else:
+        to_keep = most_recently_logged_into_account
+        to_delete = list(filter(lambda u: u is not to_keep, user_list))
+        merge_accounts(to_keep, to_delete, transfer_links=True)
+
+
+def merge_users_with_multiple_accounts_with_links(user_list):
+    """
+    We should move all links into the account with the most recent login,
+    sync registrars/orgs and then delete the other ones.
+    """
+    to_keep = None
+    to_delete = []
+    for user in user_list:
+        if user.last_login:
+            if to_keep:
+                if to_keep.last_login < user.last_login:
+                    to_delete.append(to_keep)
+                    to_keep = user
+                else:
+                    to_delete.append(user)
+            else:
+                to_keep = user
+        else:
+            to_delete.append(user)
+    merge_accounts(to_keep, to_delete, transfer_links=True)
+
+
+DUPLICATIVE_USER_SQL = '''
+  SELECT
+    perma_linkuser.id,
+    perma_linkuser.email,
+    perma_linkuser.is_active,
+    perma_linkuser.is_confirmed,
+    perma_linkuser.link_count,
+    perma_linkuser.registrar_id,
+    STRING_AGG (DISTINCT perma_linkuser_organizations.organization_id::TEXT, ',') organization_ids,
+    perma_linkuser.cached_subscription_status
+  FROM
+    perma_linkuser
+    LEFT OUTER JOIN perma_linkuser_organizations ON (
+      perma_linkuser.id = perma_linkuser_organizations.linkuser_id
+    )
+  WHERE
+    LOWER(perma_linkuser.email) in (
+      SELECT
+        LOWER(perma_linkuser.email)
+      FROM
+        perma_linkuser
+      GROUP BY
+        LOWER(perma_linkuser.email)
+      HAVING
+        COUNT(*) > 1
+    )
+  GROUP BY
+    perma_linkuser.id;
+'''
+
+def get_and_categorize_duplicative_users():
+    from perma.models import LinkUser
+
+    duplicative_users = LinkUser.objects.raw(DUPLICATIVE_USER_SQL)
+    grouped_duplicative_users = defaultdict(list)
+
+    start = time.time()
+    count = 0
+    for user in duplicative_users:
+        count += 1
+        grouped_duplicative_users[user.email.lower()].append(user)
+    end = time.time()
+
+    print(f"Data collected in {end - start} seconds.")
```

Comment:
Maybe can remove some or all these timing checks, now that we know the order of magnitude.